### DISCUSSION TITLE
Adapt core meteor/mongo package to be fibers-agnostic

### DISCUSSION THREAD

**Discussion by perbergland** [meteor/meteor#12304] [Feature Requests]
@denihs I couldn’t find a work item in [the project ](https://github.com/orgs/meteor/projects/8) for the work of moving all the implementation in meteor/mongo to be fully async-aware and fiber-agnostic.

There are 10 instances of "Npm.require('fibers" in the packages/mongo directory in meteor. Some import fibers, some import futures - all have to be taken care of to become compatible with both fiber and non-fiber meteor.

I assume this will also mean that the "sync" versions of all calls (e.g. findOne) will just invoke their async versions and fiber-await them, which will then crash with a nice error message if fibers are disabled.

`MeteorPromise.await(this.findOneAsync(...args))`

**Comment by denihs** [comment]
We did not forget those. It'll be easier to tackle those once we finish some more complicated ones like [this](https://github.com/meteor/meteor/pull/12310) or [this](https://github.com/meteor/meteor/pull/12157). That's why we don't have a task yet.
